idaes/ui/flowsheet.py

```diff
@@ -380,6 +381,31 @@ def _construct_output_json(self):
         self._construct_jointjs_json()
 
     def _construct_model_json(self):
+        from idaes.core.util.tables import create_stream_table_dataframe # deferred to avoid circular import
+
+        # Get the stream table and add it to the model json
+        # Change the index of the pandas dataframe to not be the variables
+        self._stream_table_df = (
+            create_stream_table_dataframe(self.arcs)
+                # Change the index of the pandas dataframe to not be the variables
+            .reset_index().rename(columns={"index": "Variable"})
+            .reset_index().rename(columns={"index": ""})
+            .round(5)
+        )
+
+        # Parse the names of the variables to get rid of flow_mol_phase_comp
+        self._stream_table_df['Variable'] = (
+                                                self._stream_table_df["Variable"]
+                                                .str.replace("flow_mol_phase_comp", "")
+                                                .str.rstrip()
+                                            )
+        self._stream_table_df = self._stream_table_df.where((pd.notnull(self._stream_table_df)), None)
```

Comment:
Is this line used to keep only rows without NaNs? If so, `pd.DataFrame.dropna()` could be a more succinct alternative: [pandas.DataFrame.dropna — pandas 1.2.1 documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.dropna.html#pandas-dataframe-dropna)

Reply:
No it's to replace the NaNs with None so that JSON can handle it in the ajax request. We still want the values to be there.

Reply:
Ah I see, it makes sense. I think that it could be possible to use the uncharacteristically readable `df.replace({np.nan: None})` for this, but otherwise this solutions looks fine (maybe just copying your reply as a comment, for posterity).

Reply:
I added a comment.